packages/glimmer-runtime/lib/dom/helper.ts

```diff
@@ -165,16 +196,55 @@ class DOMHelper {
   }
 }
 
+export function insertHTMLBefore(this: void, _useless: Simple.HTMLElement, _parent: Simple.Element, _nextSibling: Simple.Node, html: string): Bounds { // tslint:disable-line
+  // TypeScript vendored an old version of the DOM spec where `insertAdjacentHTML`
+  // only exists on `HTMLElement` but not on `Element`. We actually work with the
+  // newer version of the DOM API here (and monkey-patch this method in `./compat`
+  // when we detect older browsers). This is a hack to work around this limitation.
+  let parent = _parent as HTMLElement;
+  let useless = _useless as HTMLElement;
+  let nextSibling = _nextSibling as Node;
```

Comment:
@wycats Does this casting seem right? I believe this pass is only going to be taken in the browser so it seems fine to upcast these here.